pkg/channel/message_dispatcher.go

```diff
@@ -235,15 +236,20 @@ func (d *MessageDispatcherImpl) executeRequest(ctx context.Context,
 	execInfo.Time = dispatchTime
 
 	body := make([]byte, attributes.KnativeErrorDataExtensionMaxLength)
+	readLen, err := response.Body.Read([]byte(body))
+
+	// encodeBase64 encodes response body as base64 for the resulting length.
```

Comment:
I think if to long, we reduce it to first 1024

also this might need some reflection on some of the rekt/e2e tests (see failures)

Reply:
Got it